staging/src/k8s.io/component-base/cli/flag/tracker_flag.go

```diff
@@ -0,0 +1,73 @@
+/*
+Copyright 2025 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package flag
+
+import (
+	"fmt"
+
+	"github.com/spf13/pflag"
+)
+
+// TrackerValue wraps a value and stores true in the provided boolean when it is set
+type TrackerValue struct {
+	value    pflag.Value
+	provided *bool
+}
+
+var _ pflag.Value = &TrackerValue{}
+
+// NewTracker returns a Value wrapping the given value which stores true in the provided boolean when it is set
+func NewTracker(value pflag.Value, provided *bool) *TrackerValue {
+	return &TrackerValue{value: value, provided: provided}
+}
+
+func (f *TrackerValue) String() string {
+	if f.value == nil {
+		return "<nil value>"
+	}
+	return f.value.String()
+}
+
+func (f *TrackerValue) Set(value string) error {
+	if f.provided == nil {
+		return fmt.Errorf("nil pointer to provided")
+	}
+
+	err := f.value.Set(value)
+	if err == nil {
+		*f.provided = true
+	}
+	return err
+}
+
+func (f *TrackerValue) Type() string {
+	if f.value == nil {
+		return "<nil type>"
+	}
```

Comment:
it's an error to initialize an instance of this with a nil value delegate, panic in the constructor and drop the nil handling in String()/Set()/Type()